### DISCUSSION TITLE
Depreciated Options

### DISCUSSION THREAD

**Discussion by VoskSec** [rapid7/metasploit-framework#15364] [Q&A] [answered]
I noticed when taking the CyberMentor's practical ethical hacking course that in MSF5, there existed a set of options in smb_version that do not exist in MSF6.  I presume they have been depreciated, my question is what is the best way to determine WHY module options might have been depreciated?

**Comment by bcoles** [comment, marked as answer]
> my question is what is the best way to determine WHY module options might have been depreciated?

The best way is probably to search the GitHub issue tracker and pull requests.

* https://github.com/rapid7/metasploit-framework/issues?q=smb
* https://github.com/rapid7/metasploit-framework/pulls?q=is%3Apr+smb

All new changes are submitted as pull requests and then merged. You will be able to read the motivating reasons for the change and subsequent discussions.

You could also look at the [Metasploit blog](https://www.rapid7.com/blog/tag/metasploit/) which publishes weekly updates including a changelog.

If you have some familiarity with the framework you could also look through the git change history, starting with `auxiliary/scanner/smb/smb_version`:

https://github.com/rapid7/metasploit-framework/commits/master/modules/auxiliary/scanner/smb/smb_version.rb

However, changes to core libraries (ie, `Exploit::Remote::SMB`) will not be shown here. You could take a look at the git change history for `lib/msf/core/exploit/remote/smb/client.rb` :

https://github.com/rapid7/metasploit-framework/commits/master/lib/msf/core/exploit/remote/smb/client.rb

You may also need to track down dependencies. For example, change history for the `Rex::Proto::SMB` library is here:

https://github.com/rapid7/metasploit-framework/commits/master/lib/rex/proto/smb

>  I presume they have been depreciated

The options may have been renamed or change to advanced options. You can show advanced options with `show advanced`.

**Reply by VoskSec** [reply]
Appreciate the response, I'm pretty new at this so I'm still learning my way around
